cmake/Platform/JURON_Base.cmake

```diff
@@ -0,0 +1,257 @@
+  endif ()
+
+  #
+  # Code below does setup for shared libraries.  That this is done
+  # regardless of whether the platform is static or dynamic -- you can make
+  # shared libraries even if you intend to make static executables, you just
+  # can't make a dynamic executable if you use the static platform file.
+  #
+  if ( ${compiler_id} STREQUAL XL )
+    # Flags for XL compilers if we explicitly detected XL
+    set( CMAKE_SHARED_LIBRARY_${lang}_FLAGS "-qpic" )
+    set( CMAKE_SHARED_LIBRARY_CREATE_${lang}_FLAGS "-qmkshrobj -qnostaticlink" )
+  elseif ( ${compiler_id} STREQUAL LLVM )
+    set( CMAKE_SHARED_LIBRARY_${lang}_FLAGS "-std=c++11" )
+  else ()
+    # Assume flags for GNU compilers (if the ID is GNU *or* anything else).
+    set( CMAKE_SHARED_LIBRARY_${lang}_FLAGS "-fPIC" )
+    set( CMAKE_SHARED_LIBRARY_CREATE_${lang}_FLAGS "-shared" )
+  endif ()
+
+  # Both toolchains use the GNU linker on JURON, so these options are shared.
+  set( CMAKE_SHARED_LIBRARY_RUNTIME_${lang}_FLAG "-Wl,-rpath," )
+  set( CMAKE_SHARED_LIBRARY_RPATH_LINK_${lang}_FLAG "-Wl,-rpath-link," )
+  set( CMAKE_SHARED_LIBRARY_SONAME_${lang}_FLAG "-Wl,-soname," )
+  set( CMAKE_EXE_EXPORTS_${lang}_FLAG "-Wl,--export-dynamic" )
+  set( CMAKE_SHARED_LIBRARY_LINK_${lang}_FLAGS "" )  # +s, flag for exe link to use shared lib
+  set( CMAKE_SHARED_LIBRARY_RUNTIME_${lang}_FLAG_SEP ":" ) # : or empty
+
+endmacro()
+
+#
+# This macro needs to be called for dynamic library support.  Unfortunately on BG,
+# We can't support both static and dynamic links in the same platform file.  The
+# dynamic link platform file needs to call this explicitly to set up dynamic linking.
+#
+macro( __juron_setup_dynamic compiler_id lang )
+  __juron_common_setup( ${compiler_id} ${lang} )
+
+  set( CMAKE_FIND_LIBRARY_SUFFIXES ".so;.a" )
+
+  if ( ${compiler_id} STREQUAL XL )
+    set( JURON_${lang}_DYNAMIC_EXE_FLAGS "-qnostaticlink -qnostaticlink=libgcc" )
+  else ()
+    set( JURON_${lang}_DYNAMIC_EXE_FLAGS "-dynamic" )
+  endif ()
+
+  # For dynamic executables, need to provide special BG/Q arguments.
+  set( JURON_${lang}_DEFAULT_EXE_FLAGS
+      "<FLAGS> <CMAKE_${lang}_LINK_FLAGS> <LINK_FLAGS> <OBJECTS>  -o <TARGET> <LINK_LIBRARIES>" )
+  set( CMAKE_${lang}_LINK_EXECUTABLE
+      "<CMAKE_${lang}_COMPILER> -Wl,-relax ${JURON_${lang}_DYNAMIC_EXE_FLAGS} ${JURON_${lang}_DEFAULT_EXE_FLAGS}" )
+endmacro()
+
+#
+# This macro needs to be called for static builds.  Right now it just adds -Wl,-relax
+# to the link line.
+#
+macro( __juron_setup_static compiler_id lang )
+  __juron_common_setup( ${compiler_id} ${lang} )
+
+  set( CMAKE_FIND_LIBRARY_SUFFIXES ".a" )
+
+  # For static executables, use default link settings.
+  set( JURON_${lang}_DEFAULT_EXE_FLAGS
+      "<FLAGS> <CMAKE_${lang}_LINK_FLAGS> <LINK_FLAGS> <OBJECTS>  -o <TARGET> <LINK_LIBRARIES>" )
+  set( CMAKE_${lang}_LINK_EXECUTABLE
+      "<CMAKE_${lang}_COMPILER> -Wl,-relax ${JURON_${lang}_DEFAULT_EXE_FLAGS}" )
+endmacro()
```

Comment:
Some of the comments refer to Blue Gene Q. This needs to be corrected.

Reply:
@gtrensh OK. I'll remove those.